### PR TITLE
:sparkles: Add `flatten` attribute

### PR DESCRIPTION
**Pull request by elbeno**

### CODE REVIEW

File: include/stdx/compiler.hpp [hunk 1]
```diff
@@ -22,12 +22,24 @@
 #endif
 #endif
 
+#ifndef USING_ATTR_NS
+#if defined(__clang__)
+#define USING_ATTR_NS using clang:
+#else
+#define USING_ATTR_NS using gnu:
+#endif
+#endif
```
Comment (lukevalenty) [inline include/stdx/compiler.hpp:31]: Question: should this be "clang::"?

File: test/compiler.cpp
Comment (lukevalenty) [inline test/compiler.cpp:1]: question: can we combine ALWAYS_INLINE and FLATTEN? maybe we should have a test for it.

**Review by lukevalenty — APPROVED** [review]

**Review by lukevalenty — COMMENTED** [review]
Removing my approval so I can see the fixes 😅

File: include/stdx/compiler.hpp (hunk 1 above)
Comment (elbeno) [reply include/stdx/compiler.hpp:31]: No... this is selecting between the clang and gnu namespaces.

File: test/compiler.cpp
Comment (elbeno) [reply test/compiler.cpp:1]: Added a declaration.

**Review by lukevalenty — APPROVED** [review]

**Pull request merged**